NameGame/NameGame/UI/Game/GameViewController.swift

```diff
@@ -0,0 +1,124 @@
+//
+//  GameViewController.swift
+//  NameGame
+//
+//  Created by Tammy Le on 6/27/17.
+//  Copyright © 2017 WillowTree Apps. All rights reserved.
+//
+
+import UIKit
+import ReSwift
+
+protocol GameViewEventHandler {
+    func didGetUserChoice(_ choice: UserRecord)
+    func didGetNewQuestion()
+    func didEndGame()
+}
+
+class GameViewController: UIViewController {
+    @IBOutlet weak var nameLabel: UILabel!
+    @IBOutlet var userButtons: [UIButton]!
+    @IBOutlet weak var scoreLabel: UILabel!
+    
+    fileprivate var imageDictionary: [UserRecord: UIImage]!
+    
+    fileprivate var viewModel = GameViewModel()
+    weak var lifecycleObserver: ViewLifecycleObserver? = nil
+    private var eventHandler: GameViewEventHandler!
+    
+    class func make() -> GameViewController {
+        return UIStoryboard.main.make()
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        lifecycleObserver?.viewDidLoad()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        lifecycleObserver?.viewWillAppear(animated)
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        lifecycleObserver?.viewDidAppear(animated)
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        lifecycleObserver?.viewWillDisappear(animated)
+    }
+    
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        lifecycleObserver?.viewDidDisappear(animated)
+    }
+    
+    @IBAction func userButtonPressed(_ sender: UIButton) {
+        guard let selectedImage = sender.backgroundImage(for: .normal) else {
+            return
+        }
+        guard let selectedUser = imageDictionary.key(for: selectedImage) else {
```

Comment:
We should talk about another way to get the context of these buttons mapped back to our view model.